### ISSUE TITLE
No history sync between Android and Windows

### ISSUE THREAD

**Issue by eturk1** [brave/brave-browser#22513]
## Description 
  No history is sync'd between Android & Windows Brave browsers.

## Steps to Reproduce
   1. type google.com in Android & go there
   2. type msn.com in Windows & go there
   3. look at brave://history/ in each browser and note there is no history shared even after 15 minutes

## Actual result:
  no typed, no history shared between devices

## Expected result:
  typed URLs should show up on alternate device in sync chain

## Reproduces how often: 
   Easily reproduced

## Desktop Brave version:
    Brave	1.39.76 Chromium: 101.0.4951.41 (Official Build) beta (64-bit) 
    Revision	93c720db8323b3ec10d056025ab95c23a31997c9-refs/branch-heads/4951@{#904}
    OS	Windows 11 Version 22H2 (Build 22598.100)

## Android Device details:
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): Pixel 6 Pro
   - Android version: 12

## Other Additional Information:
  The setting "Automatically sync app data" is turned on.

  Note: the settings stated in the FAQ for help with Android do not exist on Pixel 6 Android 12. There is a "Passwords & accounts" in system settings w/ the setting name above.
(Maybe it's an old tips and not testing on Android 12?) https://support.brave.com/hc/en-us/articles/360047642371-Sync-FAQ

**Comment by bsclifton** [comment]
This is a known limitation. For history, I believe only things you type into the URL box are synced. The entire history is not synced (not even in Chrome)

cc: @AlexeyBarabash for confirmation

**Comment by AlexeyBarabash** [comment]
Yes, for history not all the entries are synced one-to-one in Chromium and Brave.
Here are more details https://github.com/brave/brave-browser/issues/13293#issuecomment-788877691 .

I have checked on current Beta build Android <=> Linux desktop - the history synced well for me.

I am sure history entries will be synced if:
- on a desktop to make the visit be synced, the url should be typed and then clicked by mouse on the top occurrence in dropdown;
- on Android to make entry be synchronized, url should be typed or pasted and then Enter must be pressed on the virtual keyboard.

It is the behavior inherited from Chromium. We thought about turning syncing of all history entries, but we found mentions that it caused out of memory errors while syncing browser with a huge and intensive history.
